### PR TITLE
Implement August/September Seasonal, Rainbow Road

### PR DESCRIPTION
**Pull request by ascriptmaster**
Rainbow Road is a Mario Kart and color themed seasonal.

Every few turns, a player can get an item that changes the course of the
battle, such as a Mushroom to increase speed.

Pokemon also get color moves that are super-effective on certain other
colors.

----

Something something. Most of this should be tested but I can't be sure about things like the Blooper lowering accuracy or whatever.

Messages at the start of the battle could be improved.

@Zarel @Slayer95

### CODE REVIEW

**Comment by Joimer** [comment]
wow, this is a great idea! The Travis build is failing, though, can you check that?

**Comment by ascriptmaster** [comment]
Apparently my local `npm test` doesn't want to check for trailing whitespace?

Anyways, what do we do with the start-of-battle messages? I really do feel like they could be improved

**Comment by Joimer** [comment]
I think they are informational enough, you could bold the most important parts, but I think this should do, don't you?

**Comment by ascriptmaster** [comment]
Lemme give it a day to let @Zarel, @TheImmortal and @Slayer95 respond

**Comment by Irraquated** [comment]
If you're interested in shortening you could always use `/data [POKEMON]` over `/details [POKEMON]` but other than that the start messages are very clear on how to play.

**Comment by ascriptmaster** [comment]
`/data [POKEMON]` doesn't show Pokedex color though, and I don't want to put `/dt`

File: config/formats.js [hunk 1]
```diff
@@ -481,6 +481,196 @@ exports.Formats = [
...
+							decision = this.queue[i];
+							decision.speed = pokemon.getStat('spe');
+							decisions.push(decision);
+							// Cancel the decision
+							this.queue.splice(i, 1);
+							i--;
+						}
+					}
+					for (i = 0; i < decisions.length; i++) {
+						this.insertQueue(decisions[i]);
```
Comment (Slayer95) [inline config/formats.js]: Interesting

Comment (ascriptmaster) [inline config/formats.js]: Yes, it is my new function from #2035

**Comment by TheImmortal** [comment]
>Most of this should be tested

I think you're kinda rushing it. Can it be tested/improved for the rest of the month then added for August/September?

**Comment by ascriptmaster** [comment]
There's not much else to test, honestly, only proving that the Blooper actually lowers accuracy? That should work properly, right?

Also apparently looking at it again there's a bug in the Golden Mushroom expiring message that I'll have to fix.

I guess I'm rushing it, but I think @Joimer has programmed seasonals in less time it took for me to build this one, which took 2 days. But I guess it could be tested on joimslab first

**Comment by TheImmortal** [comment]
Yeah but Joim gave staff time to test it as well.

**Comment by ascriptmaster** [comment]
Fine, I'll buy a VPS

**Comment by Zarel** [comment]
@ascriptmaster, you don't need to, you can test on any of my servers.

**Comment by Zarel** [comment]
Including smogtours, if you want.

**Comment by ascriptmaster** [comment]
I'm using Cloud9 right now, which is free. There have only been like 4-5 battles on the server right now, though.

I don't want to leak seasonals to smogtours anyhow

**Comment by Joimer** [comment]
Well, philosophy is that it can be leaked to anyone willing to help test it thoroughly, so that wouldn't be a problem.

**Comment by ascriptmaster** [comment]
New version with a different color chart and volatiles.

**Comment by TheImmortal** [comment]
For once we can have the seasonal up on the 1st!

(Hopefully this can continue for future ones)

**Comment by Joimer** [comment]
Because we changed the first month :P

File: data/statuses.js [hunk 2]
```diff
@@ -259,6 +259,7 @@ exports.BattleStatuses = {
 	},
 	choicelock: {
 		onStart: function (pokemon) {
+			console.log(this.activeMove.id);
```
Comment (ascriptmaster) [inline data/statuses.js]: mfw

**Comment by ascriptmaster** [comment]
Three days left. Can we get this testing in smogtours or something so I can get more input? I'm too busy to maintain my clunky C9 server right now and keep it online.

**Comment by ascriptmaster** [comment]
soon

**Pull request merged**